### PR TITLE
Add new types

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin @sdroege

### CODE REVIEW

File: src/auto/proxy_address.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        let password = password.into();
+        let password = password.to_glib_none();
+        unsafe {
+            SocketAddress::from_glib_full(ffi::g_proxy_address_new(inetaddr.to_glib_none().0, port, protocol.to_glib_none().0, dest_hostname.to_glib_none().0, dest_port, username.0, password.0)).downcast_unchecked()
+        }
+    }
+}
+
+pub trait ProxyAddressExt {
+    fn get_destination_hostname(&self) -> Option<String>;
```
Comment (sdroege) [inline src/auto/proxy_address.rs]: This can't be `NULL` AFAIU

File: src/auto/proxy_address.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    fn get_destination_port(&self) -> u16;
+
+    #[cfg(any(feature = "v2_34", feature = "dox"))]
+    fn get_destination_protocol(&self) -> Option<String>;
+
+    fn get_password(&self) -> Option<String>;
+
+    fn get_protocol(&self) -> Option<String>;
```
Comment (sdroege) [inline src/auto/proxy_address.rs]: Neither can this be `NULL`

**Comment by sdroege** [comment]
Looks good apart from the above. I thought I added the proxy things with the sockets back then, seems I started and forgot to finish that :)

File: src/auto/native_volume_monitor.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+use VolumeMonitor;
+use ffi;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct NativeVolumeMonitor(Object<ffi::GNativeVolumeMonitor, ffi::GNativeVolumeMonitorClass>): VolumeMonitor;
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/auto/native_volume_monitor.rs]: Ah interesting. I was able to find it in the gnome docs but it basically does nothing. I'll remove it.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/auto/proxy_address.rs [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    }
+}
+
+pub trait ProxyAddressExt {
+    fn get_destination_hostname(&self) -> String;
+
+    fn get_destination_port(&self) -> u16;
+
+    #[cfg(any(feature = "v2_34", feature = "dox"))]
+    fn get_destination_protocol(&self) -> Option<String>;
```
Comment (EPashkin) [inline src/auto/proxy_address.rs]: This IMHO also not-nullable

**Comment by EPashkin** [comment]
👍

**Comment by sdroege** [comment]
ok :)

**Pull request merged**